### ISSUE TITLE
Enhancement Request : HyperLink Label [ New Widget ]

### ISSUE THREAD

**Issue by YoussefEssalhi** [TomSchimansky/CustomTkinter#1855]
**Use Case**: I Think this enhancement would be immensely useful for developers who need to display URLs, direct users to external resources, or provide interactive elements within the application.

**Thanks**

**Comment by Akascape** [comment]
A very simple implementation of hyperlink label:
```python
import customtkinter
import webbrowser

root = customtkinter.CTk()
link = "https://github.com/TomSchimansky/CustomTkinter"
label = customtkinter.CTkLabel(root, text="customtkinter webpage", font=("",13), text_color="blue")
label.pack(padx=10, pady=10)

label.bind("<Button-1>", lambda event: webbrowser.open_new_tab(link))
label.bind("<Enter>", lambda event: label.configure(font=("",13,"underline"), cursor="hand2"))
label.bind("<Leave>", lambda event: label.configure(font=("",13), cursor="arrow"))
root.mainloop()
```

**Comment by Akascape** [comment]
@YoussefEssalhi

**Comment by YoussefEssalhi** [comment]
@Akascape The problem with this solution is that it causes a design problem

**Until you understand the problem**
suppose I want to create a text at the very bottom of an app. It typically contains a copyright notice and a link to my website

- The text alone should take a hyperlink but as you can see all the background takes it !

**Problem :**

https://github.com/TomSchimansky/CustomTkinter/assets/76540840/786f46a6-e4e0-400a-93e7-9240f60388b6

**Code :**

```python
# Packages
from customtkinter import CTk, CTkLabel
from webbrowser import open_new_tab

# Setup Window
app = CTk()
app.title("HyperLink")
app.geometry("550x350")

# HyperLink
link = "https://github.com/TomSchimansky/CustomTkinter"
label = CTkLabel(app, text="customtkinter webpage", font=('', 13), text_color="#4fa5e2", anchor='s')
label.bind("<Button-1>", lambda event: open_new_tab(link))
label.bind("<Enter>", lambda event: label.configure(font=('', 13, "underline"), cursor="hand2"))
label.bind("<Leave>", lambda event: label.configure(font=('', 13), cursor="arrow"))

# Layout
label.pack(pady=10, expand=True, fill='y')

# Run
app.mainloop()
```

That's why I think it should be **New Widget** in `CTK` as solve this problem.

**Comment by YoussefEssalhi** [comment]
**Thanks dude**

**Comment by YoussefEssalhi** [comment]
@dishb as you see, @Akascape He directly edit on `_label` did you have anything to add or leave it as is !

Because you say : "`It's not recommended to edit the attribute as it's not intended for users`"

**Comment by dishb** [comment]
@YoussefEssalhi - I still stand by that. Following Python conventions, whenever developing a library/framework, if you don't intend on having an attribute, class, variable, or function (etc.) to be exposed to users, you prefix the name with an underscore. "_". Some IDEs (in their autocompletion windows) won't show these variables/etc. at the top. However, it is possible to use these.

When I said it's not recommended, I usually feel that an issue/PR should be opened to add this change where it is intended for users to use. For the same reason I call these "workarounds" as they are temporary and not intended features.

**Comment by YoussefEssalhi** [comment]
@dishb - Now what do you advise !?

 - Open a new issue
 
 -  Reopen this issue

 or just work with this temporary solve !

**Comment by YoussefEssalhi** [comment]
@dishb - Now what do you advise !?

 - Open a new issue
 
 -  Reopen this issue

 or just i work with this temporary solve !

**Comment by YoussefEssalhi** [comment]
@dishb - Now what do you advise !?

 - Open a new issue
 
 -  Reopen this issue

 or just i work with this temporary solution !!!

**Comment by dishb** [comment]
Temporary solution, open an issue if you haven't yet :)

**Comment by YoussefEssalhi** [comment]
I'm very satisfied with solution but I'm thinking of other users how they can take advantage of the feature in an easy way

**Comment by Akascape** [comment]
I am pretty sure this will not be added as a new widget, instead there should be a new parameter for hyperlink within ctklabel.

**Comment by avalon60** [comment]
Does [this](https://www.tutorialspoint.com/how-to-create-hyperlink-in-a-tkinter-text-widget#:~:text=Tkinter%20Text%20widgets%20are%20generally,using%20HyperLinkManager%20snippet%20in%20Python.) not suffice?

**Comment by avalon60** [comment]
> @YoussefEssalhi - I still stand by that. Following Python conventions, whenever developing a library/framework, if you don't intend on having an attribute, class, variable, or function (etc.) to be exposed to users, you prefix the name with an underscore. "_". Some IDEs (in their autocompletion windows) won't show these variables/etc. at the top. However, it is possible to use these.
> 
> When I said it's not recommended, I usually feel that an issue/PR should be opened to add this change where it is intended for users to use. For the same reason I call these "workarounds" as they are temporary and not intended features.

I agree with @dishb. It is risky referencing such variables (or even methods), and can introduce a problem, waiting to happen.

**Comment by YoussefEssalhi** [comment]
> Does [this](https://www.tutorialspoint.com/how-to-create-hyperlink-in-a-tkinter-text-widget#:~:text=Tkinter%20Text%20widgets%20are%20generally,using%20HyperLinkManager%20snippet%20in%20Python.) not suffice?

This hyperlink in `Text widget`, and i'm talk about hyperlink in `label`.

**Comment by dishb** [comment]
I think they are the same.

**Comment by Akascape** [comment]
@YoussefEssalhi 
I found this ready made link widget inside ttkwidgets: https://github.com/TkinterEP/ttkwidgets/blob/master/ttkwidgets/linklabel.py

**Comment by YoussefEssalhi** [comment]
> I think they are the same.

- Hyperlink in `label`.

![image](https://github.com/TomSchimansky/CustomTkinter/assets/76540840/6de6970b-c32a-44fe-a9e6-56ddda9e174b)

- Hyperlink in `Text`

![image](https://github.com/TomSchimansky/CustomTkinter/assets/76540840/c695e96e-3ed8-478e-ab60-43ff279dbb4a)

@dishb are u sure !?

**Comment by YoussefEssalhi** [comment]
> @YoussefEssalhi I found this ready made link widget inside ttkwidgets: https://github.com/TkinterEP/ttkwidgets/blob/master/ttkwidgets/linklabel.py

**Thanks dude**, But I'm already using your first solution.

I just left the issue still open for @TomSchimansky to  add this feature to other users.
